packages/editor-ui/src/stores/root.store.ts

```diff
@@ -0,0 +1,225 @@
+import { CLOUD_BASE_URL_PRODUCTION, CLOUD_BASE_URL_STAGING, STORES } from '@/constants';
+import type { RootState } from '@/Interface';
+import { setGlobalState } from 'n8n-workflow';
+import { defineStore } from 'pinia';
+import { computed, ref } from 'vue';
+
+const { VUE_APP_URL_BASE_API } = import.meta.env;
+
+export const useRootStore = defineStore(STORES.ROOT, () => {
+	const state = ref({
+		baseUrl: VUE_APP_URL_BASE_API ?? window.BASE_PATH,
+		restEndpoint:
+			!window.REST_ENDPOINT || window.REST_ENDPOINT === '{{REST_ENDPOINT}}'
+				? 'rest'
+				: window.REST_ENDPOINT,
+		defaultLocale: 'en',
+		endpointForm: 'form',
+		endpointFormTest: 'form-test',
+		endpointFormWaiting: 'form-waiting',
+		endpointWebhook: 'webhook',
+		endpointWebhookTest: 'webhook-test',
+		pushConnectionActive: true,
+		timezone: 'America/New_York',
+		executionTimeout: -1,
+		maxExecutionTimeout: Number.MAX_SAFE_INTEGER,
+		versionCli: '0.0.0',
+		oauthCallbackUrls: {},
+		n8nMetadata: {},
+		pushRef: Math.random().toString(36).substring(2, 15),
+		urlBaseWebhook: 'http://localhost:5678/',
+		urlBaseEditor: 'http://localhost:5678',
+		isNpmAvailable: false,
+		instanceId: '',
+		binaryDataMode: 'default',
+	});
+
+	// ---------------------------------------------------------------------------
+	// #region Computed
+	// ---------------------------------------------------------------------------
+
+	const getBaseUrl = computed(() => state.value.baseUrl);
+
+	const getFormUrl = computed(() => `${state.value.urlBaseWebhook}${state.value.endpointForm}`);
+
+	const getFormTestUrl = computed(
+		() => `${state.value.urlBaseEditor}${state.value.endpointFormTest}`,
+	);
+
+	const getFormWaitingUrl = computed(
+		() => `${state.value.baseUrl}${state.value.endpointFormWaiting}`,
+	);
+
+	const getWebhookUrl = computed(
+		() => `${state.value.urlBaseWebhook}${state.value.endpointWebhook}`,
+	);
+
+	const getPushRef = computed(() => state.value.pushRef);
+
+	const getBinaryDataMode = computed(() => state.value.binaryDataMode);
```

Comment:
@alexgrozav  Should we be using getters to return single properties? or access the state directly? It seems cleaner this way but more verbose as we need to define a computed method for each property we want to access and method to mutate that property. 🤔

Reply:
Probably having individual refs would be more in line with the other composition api stores. We could have refs directly instead of a `state` object. This would remove the need for writing individual getters for each property and you'd be able to mutate the property value directly without a setter:

```
// ChatGPT Generated
import { ref } from 'vue';
.... 
const baseUrl = ref(VUE_APP_URL_BASE_API ?? window.BASE_PATH);
const restEndpoint = ref(
  !window.REST_ENDPOINT || window.REST_ENDPOINT === '{{REST_ENDPOINT}}'
    ? 'rest'
    : window.REST_ENDPOINT
);
const defaultLocale = ref('en');
const endpointForm = ref('form');
const endpointFormTest = ref('form-test');
const endpointFormWaiting = ref('form-waiting');
const endpointWebhook = ref('webhook');
const endpointWebhookTest = ref('webhook-test');
const pushConnectionActive = ref(true);
const timezone = ref('America/New_York');
const executionTimeout = ref(-1);
const maxExecutionTimeout = ref(Number.MAX_SAFE_INTEGER);
const versionCli = ref('0.0.0');
const oauthCallbackUrls = ref({});
const n8nMetadata = ref({});
const pushRef = ref(Math.random().toString(36).substring(2, 15));
const urlBaseWebhook = ref('http://localhost:5678/');
const urlBaseEditor = ref('http://localhost:5678');
const isNpmAvailable = ref(false);
const instanceId = ref('');
const binaryDataMode = ref('default');
```

It's a matter of preference really, with your approach we're essentially making sure we don't modify a value by mistake, but there will be a bit more maintenance. Given the importance of this store, we should probably keep it this way.

Anyhow, for better DX, I'd use variable names without `get`, since it's a computed property not a function, and when it's used it's more natural to write `rootStore.baseUrl`.